services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/service/implement/SubnetServiceImp.java

```diff
@@ -196,4 +205,20 @@ public IpAddrRequest allocateIPGateway(String subnetId) throws FallbackException
 
         return ipResponse;
     }
+
+    @Override
+    public String[] cidrToFirstIpAndLastIp(String cidr) {
+        if (cidr == null) {
+            return null;
+        }
+        SubnetUtils utils = new SubnetUtils(cidr);
+        String[] address = utils.getInfo().getAllAddresses();
```

Comment:
getAllAddress could be very costly for a huge subnet. Could we do it more efficient?

Reply:
Yes ~ you are right